www/jxcore/app.js

```diff
@@ -124,29 +152,262 @@ Mobile('stopThali').registerSync(function () {
-                console.log("TestApp error while adding data: " + err);
-            });
+});
+
+var attachmentIndex = 0;
+
+Mobile('addAttachment').registerSync(function () {
+    var attachment = new Buffer('attachment/attachment:' + attachmentIndex + ':' + new Date());
+    attachmentIndex ++;
+
+    localDB
+    .putAttachment(
+      attachment.toString(),
+      'attachment',
+      attachment,
+      'text/plain'
+    )
+        .then(function () {
+            console.log('sent attachment');
+        })
+        .catch(function (error) {
+            console.error('got error: \'%s\'', error);
+        });
+});
+
+
+var DOCS_COUNT             = 60;
+var DOC_SEND_TIMEOUT       = 1000;
+var DOC_SEARCH_TIMEOUT     = 2 * 60 * 1000;
+var NETWORK_TOGGLE_TIMEOUT = 1 * 60 * 1000;
+var SILENCE_TIMEOUT        = 4 * 60 * 1000;
+
+/*
+var TIMES_FASTER = 12;
+DOC_SEND_TIMEOUT       = Math.round(DOC_SEND_TIMEOUT / TIMES_FASTER);
+NETWORK_TOGGLE_TIMEOUT = Math.round(NETWORK_TOGGLE_TIMEOUT / TIMES_FASTER);
+SILENCE_TIMEOUT        = Math.round(SILENCE_TIMEOUT / TIMES_FASTER);
+*/
+
+
+function waitForRemoteDocs(pouchDB, round, docsCount) {
+  function allDocsFound() {
+    // We want to find at least 'docsCount' documents.
+    return docsCount === 0;
+  }
+
+  function verifyDoc(doc) {
+    if (
+      doc.deviceId === undefined || doc.deviceId === null ||
+      doc.round    === undefined || doc.round    === null
+    ) {
+      // console.log('this is not our doc');
+    } else if (doc.deviceId === myDeviceId) {
+      // console.log('local doc found');
+    } else if (doc.round === round) {
+      docsCount--;
+      console.log('remote doc found, %d docs remaining', docsCount);
     }
-});
\ No newline at end of file
+  }
+
+  return new Promise(function (resolve, reject) {
+    var error;
+    var completed = false;
+    function complete () {
+      if (completed) {
+        return;
+      }
+      completed = true;
+
+      if (error) {
+        reject(error);
+      } else {
+        resolve();
+      }
+    }
+    var changesFeed = pouchDB.changes({
+      live: true,
+      include_docs: true
+    })
+      .on('change', function (change) {
+        verifyDoc(change.doc);
+        if (allDocsFound()) {
+          changesFeed.cancel();
+        }
+      })
+      .on('complete', function (info) {
+        if (info.errors && info.errors.length > 0) {
+          error = info.errors[0];
+        }
+        complete();
+      })
+      .on('error', function (err) {
+        error = err;
+        complete();
+      });
+  });
+}
+
+function native (target, value) {
+  return new Promise(function (resolve, reject) {
+    Mobile(target).callNative(value, function (error) {
+      if (error) {
+        reject(error);
+      } else {
+        resolve();
+      }
+    });
+  });
+}
+
+function sendData (round, wantToggleWiFi, wantToggleBluetooth) {
+  console.log('sending new docs');
+
+  function send(attempts, timeout) {
```

Comment:
To me the variable name attempts implies that we are attempting to perform an action and if it fails then we will try again. But the code looks more like it is just repeating an action a fixed number of times. So 'docsToCreate' might be a more accurate name.